### DISCUSSION TITLE
Polite discussion about Map Mode selector of the Live Map

### DISCUSSION THREAD

**Discussion by eickeh** [Hypfer/Valetudo#1747] [Q&A / Support] [answered]
Hi there,

First of all thanks for all your great work! I'm happily sending my robot out several times daily to sweep up the remains of my toddlers food and am really glad I don't have to do that myself and especially without the manufacturer watching :D 

However, since the introduction of the Map Mode selector of the Live Map, I have a really hard time getting used to this and hope this thread can get a discussion about how would be the best way do arrange those features. I understand the idea behind the Mode selector and can appreciate that in a way. However, it's not intuitive for me :D 

Let me explain:
- I use the zone cleaning a lot and also let my robot clean the entrance segment very often. On some days both several times a day. Before the change, I could just do either of those with one click, now I need two clicks more to switch between zone and segment cleaning. 
- For me, it's very counter-intuitive (I don't seem to get used to it...) so when I'm in zone cleaning mode and want to clean a segment I tap several times on the segment before noticing "argh, wrong mode again".

My suggestion would be the following:
- Zone and segment cleaning are handled as before (was perfect for me): Zone cleaning buttons are visible at all times and you're able to select a segment by clicking on it anywhere
- Place a "Go to" Button in some corner. To send it somewhere, click the go to button, then click on the map and click send or cancel. Additionally, one could put the go to on long press for users that need this feature more often.

What do you think?

**Comment by Hypfer** [comment, marked as answer]
https://xkcd.com/1172/

Apart from that, I do of course understand that UI changes are always frustrating especially if you're used to how it was before but what can you do. It just happens sometimes.

The suggestion you provided would work for you but would definitely not work for new users, as then you'd have two modes.
- One that is a single mode (GoTo) 
- Another that is effectively two modes as one mode.

It would be inconsistent and confusing to do it like that, because why should one mode be two modes


Another aspect of that new mode switch is discoverability. 
Not only does it provide a way to switch modes but it also clearly shows you a list of things you can do with the map view. If you combine segments and zones again, the user has to resort to trial and error to figure that out.

I'm sure you'll get used to how it is now in a while. It will be a bit annoying until then but again that's just how it is.
Things do sometimes have to change. At least here there's a good reason for that and not just google-style changing **everything** every two years for the sake of doing so.

**Comment by eickeh** [comment]
Haha, answering with an xkcd - I like that =) I actually had this one in mind during writing - That's why I abstained from asking for an option to revert the old behavior^^

So, keeping it new-user-friendly. The picture below shows a quick draft of another suggestion (not pretty but shows the idea):
![valetudo_design_suggestion](https://user-images.githubusercontent.com/33962585/225156114-4f281f75-2811-4553-a52d-1719ab4d10ce.jpg)
The hovering selector would be replaced by a horizontal selection directly stating the modes. Switching modes could be done by clicking or swiping. This would keep the UI clean and directly show the possible options. New users directly see it and do not even have to discover the floating map selector button.  
And I would be happy as there is just one extra click and I directly see in which mode I am. :D

What about something like that?

**Reply by Hypfer** [reply]
A few question come to mind
- How would that look like in the non-mobile view?
- How do you manage to tap on such tiny controls on mobile? And if you enlarge them, how much space remains?
- What happens on very small mobile screens where the bottom buttons get pushed into more than one row?

Also, generally, there would be a sandwich of three rows of controls at the bottom of the screen which is a bit too heavy for my taste.

Another note:
Swiping is actually quite hard to do reliably in a browser. The libraries I've tried so far were also very resource-intensive for some reason.


> I directly see in which mode I am

You also know that by looking at the icon at the top. Granted, they're not ideal right now as they're generic mui icons and not ones specifically designed for Valetudo but it would be possible to have such icons

**Reply by eickeh** [reply]
I see, discussion closed. Thanks anyway for your time!
Maybe I'll just try installing the previous version manually with the user interface that gives me wet dreams :D
